### ISSUE TITLE
fail to compile because of 'ar crs'

### ISSUE THREAD

**Issue by a-bad-apple** [alexcrichton/cc-rs#504]
My environment are Windows10, Gnu and busybox.

--- stdout
cargo:rerun-if-env-changed=LIBZ_SYS_STATIC
cargo:rerun-if-changed=build.rs
TARGET = Some("x86_64-pc-windows-gnu")
OPT_LEVEL = Some("3")
HOST = Some("x86_64-pc-windows-gnu")
CC_x86_64-pc-windows-gnu = None
CC_x86_64_pc_windows_gnu = None
HOST_CC = None
CC = None
CFLAGS_x86_64-pc-windows-gnu = None
CFLAGS_x86_64_pc_windows_gnu = None
HOST_CFLAGS = None
CFLAGS = None
CRATE_CC_NO_DEFAULTS = None
DEBUG = Some("false")
CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")

cargo:warning=BusyBox v1.32.0-FRP-3329-gcf0fa4d13 (2020-01-09 09:35:27 GMT) multi-call binary
cargo:warning=
cargo:warning=Usage: ar [-o] [-v] [-p] [-t] [-x] ARCHIVE FILES
cargo:warning=
cargo:warning=Extract or list FILES from an ar archive
cargo:warning=
cargo:warning=  -o      Preserve original dates
cargo:warning=  -p      Extract to stdout
cargo:warning=  -t      List
cargo:warning=  -x      Extract
cargo:warning=  -v      Verbose
cargo:warning=F:\scoop\apps\busybox\current\busybox.exe: unknown option -- s
exit code: 1

--- stderr

The fellow is output of ar:
>ar crs
F:\scoop\apps\busybox\current\busybox.exe: unknown option -- s
BusyBox v1.32.0-FRP-3329-gcf0fa4d13 (2020-01-09 09:35:27 GMT) multi-call binary

Usage: ar [-o] [-v] [-p] [-t] [-x] ARCHIVE FILES

Extract or list FILES from an ar archive

        -o      Preserve original dates
        -p      Extract to stdout
        -t      List
        -x      Extract
        -v      Verbose

The 's' option is invalid in windows10.

**Comment by alexcrichton** [comment]
Thanks for the report! It looks like this is a nonstandard implementation of `ar`? Is this perhaps a bug in your `ar` executable?

**Comment by yashraghaven** [comment]
Hi! I faced the same issue while trying to build another rust application. Busybox's "ar" is not a full implementation (a comment in the source says "Mini ar implementation for busybox"). I have mingw-w64 installed, there's a full ar implementation there. I made the mingw-w64 bin directory available in the Windows path ahead of the busybox path and that solved the problem for me.

**Comment by joshtriplett** [comment]
Yeah, busybox `ar` isn't designed to create archives, only extract them. It isn't a usable implementation for compilation.
